packages/graphql-language-service-server/src/startServer.js

```diff
@@ -17,67 +17,88 @@ import {
 } from './MessageProcessor';
 
 import {
+  IPCMessageReader,
   IPCMessageWriter,
   SocketMessageReader,
   SocketMessageWriter,
 } from 'vscode-jsonrpc';
 
-type ServerOptions = {
+type Options = {
   port?: number,
+  method?: string,
+  configDir?: string,
 };
 
-export default (async function startServer(
-  configDir: ?string,
-  options?: ServerOptions,
-): Promise<void> {
-  if (options && options.port) {
-    // Socket protocol support
-    const socket = net.createServer(client => {
-      client.setEncoding('utf8');
-      const messageReader = new SocketMessageReader(client);
-      const messageWriter = new SocketMessageWriter(client);
+export default (async function startServer(options?: Options): Promise<void> {
+  if (!options || !options.configDir) {
+    process.stderr.write('--configDir is required!');
+    process.exit(1);
+  }
+
+  const configDir = options.configDir;
+  if (options && options.method) {
```

Comment:
I gather `options &&` here is just to keep Flow happy? (Seeing as it can't be `null`-ish here.)